### PR TITLE
feat(WIP): web support based on IndexedDB

### PR DESCRIPTION
**Pull request by adhorodyski**

### CODE REVIEW

File: src/lib/setupDatabaseHook.web.ts [hunk 1]
```diff
@@ -1,11 +1,17 @@
...
-import {db, type Row} from './db';
+import {db} from './db.web';
 import {databaseQueryHook} from './databaseQueryHook';
 
 export const setupDatabaseHook = (queryClient: QueryClient) => {
-  // TODO: this is a placeholder for the actual implementation that calls `databaseQueryHook()`
-  db.updateHook(params => {
-    const keyChanged = (params.row as Row).key;
+  db.kv.hook('creating', keyChanged => {
+    databaseQueryHook(keyChanged, queryClient);
```
Comment (adhorodyski) [inline src/lib/setupDatabaseHook.web.ts]: FYI @jbroma so far each hook is getting called correctly, but our `databaseQueryHook` somehow fails from time to time and doesn't feel reliable (on web only which makes it even more weird, it's the same, generic implementation).

TLDR i have more console logs (triggered correctly here) than UI renders based on my in-app actions.

Comment (jbroma) [reply src/lib/setupDatabaseHook.web.ts]: it's fixed already right?

Comment (jbroma) [reply src/lib/setupDatabaseHook.web.ts]: what about `Dexie.on.storagemutated` event, isn't that even simpler?

**Comment by jbroma** [comment]
@adhorodyski what's the status of this, perhaps we can merge it already? I've tested this locally and all seems to be working well, even multitab sync
